packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts

```diff
@@ -18,11 +18,13 @@ const CAPITALIZE_INITIALISM: generatorsYml.GenerationLanguage[] = ["go", "ruby"]
 export function constructCasingsGenerator({
     generationLanguage,
     keywords,
-    smartCasing
+    smartCasing,
+    casingVersion
 }: {
     generationLanguage: generatorsYml.GenerationLanguage | undefined;
     keywords: string[] | undefined;
     smartCasing: boolean;
+    casingVersion: generatorsYml.CasingVersion | undefined;
```

Comment:
do you think we can delete smartCasing altogether and just rely on `casingVersion` now  here? 

i would have expected the code to read something like: 
```ts
switch (version) {
  case "v0": 
    // do a vanilla case
  case "v1": 
    // handle initialisms and numbers
  default: 
    assertNever(version)
}
```

Reply:
So you want to have a yaml migration to move smartCasing to v0?

Reply:
I think internally in the codebase we can have `casingVersion` (in `ir-generator` for example), but externally we can listen to version first and then fall back to smart-casing -> v1

Reply:
so no yaml migration needed